src/DropDown/DropDown.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useCallback, useMemo } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[];
+  selectedPerson: Person | null;
+  onSelect?: (person: Person) => void;
+  delay: number
+};
+
+export const DropDown: React.FC<Props> = React.memo(({
+  people,
+  selectedPerson,
+  onSelect = () => {},
+  delay,
+}) => {
+  const [query, setQuery] = useState('');
+  const [apliedQuery, setApliedQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = useCallback(debounce(setApliedQuery, delay), []);
+
+  const visiblePeople = useMemo(() => (people.filter(person => {
+    const normalizedQuery = apliedQuery.trim().toLowerCase();
+
+    return person.name.trim().toLowerCase().includes(normalizedQuery);
+  })), [apliedQuery, people]);
+
+  const resultToShow = (visiblePeople.length > 0);
+
+  const isPeopleListShown = !selectedPerson
+  && resultToShow
+  && isFocused;
+
+  const queryHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const selectHander = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    person: Person,
+  ) => {
+    event.preventDefault();
+    onSelect(person);
+    setIsFocused(false);
+    setQuery('');
+  };
+
+  return (
+    <div className="dropdown is-active">
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onChange={queryHandler}
+          onClick={() => setIsFocused(true)}
+          onBlur={() => setIsFocused(false)}
+          disabled={selectedPerson !== null}
```

Comment:
This is a bit strange from the point of the user, so I am not sure of it is a good idea to do so